server/jetstream_batching.go

```diff
@@ -41,26 +42,71 @@ type batchGroup struct {
 	timer *time.Timer
 }
 
-func newBatchGroup(s *Server, batchId string, batches *batching, store StreamStore) *batchGroup {
-	b := &batchGroup{store: store}
+// Lock should be held.
+func (batches *batching) newBatchGroup(mset *stream, batchId string) (b *batchGroup, err error) {
```

Comment:
Let's avoid named returns here unless they are used exclusively, which it looks like they aren't.

Reply:
Done